crmsh/bootstrap.py

```diff
@@ -1739,35 +1794,24 @@ def join_ssh(seed_host):
         error("No existing IP/hostname specified (use -c option)")
 
     start_service("sshd.service")
-    invoke("mkdir -m 700 -p /root/.ssh")
-
-    tmpdir = tmpfiles.create_dir()
-    status("Retrieving SSH keys - This may prompt for root@%s:" % (seed_host))
-    if not invoke("scp -oStrictHostKeyChecking=no  root@%s:'/root/.ssh/id_*' %s/" % (seed_host, tmpdir)):
-        error("Failed to retrieve ssh keys")
-
-    # This supports all SSH key types, for the case where ha-cluster-init
-    # wasn't used to set up the seed node, and the user has manually
-    # created, for example, DSA keys (bnc#878080)
-    got_keys = 0
-    for key in ("id_rsa", "id_dsa", "id_ecdsa", "id_ed25519"):
-        if not os.path.exists(os.path.join(tmpdir, key)):
-            continue
-
-        if os.path.exists(os.path.join("/root/.ssh", key)):
-            if not confirm("/root/.ssh/%s already exists - overwrite?" % (key)):
-                continue
-        invoke("mv %s* /root/.ssh/" % (os.path.join(tmpdir, key)))
-        if not grep_file("/root/.ssh/authorized_keys", open("/root/.ssh/%s.pub" % (key)).read()):
-            append("/root/.ssh/%s.pub" % (key), "/root/.ssh/authorized_keys")
-        got_keys += 1
-
-    if got_keys == 0:
-        status("No new SSH keys installed")
-    elif got_keys == 1:
-        status("One new SSH key installed")
-    else:
-        status("%s new SSH keys installed" % (got_keys))
+    configure_local_ssh_key()
+
+    # Detect whether need password to login to seed_host
+    passwordless = False
```

Comment:
Why do you use this `passwordless` variable? This `if/else` condition looks super strange

Why not simply 
```
if not utils.check_ssh_passwd_need(seed_host):
    status("Configuring SSH keys - This may prompt for root@{}".format(seed_host))
    ...
```

Reply:
Thanks for reminding this, I thought that `passwordless` would be used multi times

Already changed